packages/inline-edit/src/InlineEdit.tsx

```diff
@@ -0,0 +1,137 @@
+import type { InlineVariants } from './styles/InlineEdit.css';
+import type { TextAreaProps, TextFieldProps } from '@launchpad-ui/form';
+import type {
+  ComponentProps,
+  Dispatch,
+  KeyboardEventHandler,
+  ReactElement,
+  SetStateAction,
+} from 'react';
+
+import { ButtonGroup, IconButton } from '@launchpad-ui/button';
+import { TextField } from '@launchpad-ui/form';
+import { Icon } from '@launchpad-ui/icons';
+import { useButton } from '@react-aria/button';
+import { focusSafely } from '@react-aria/focus';
+import { mergeProps, useUpdateEffect } from '@react-aria/utils';
+import { cx } from 'classix';
+import { cloneElement, useRef, useState } from 'react';
+
+import { container, cancelButton, inline, readButton } from './styles/InlineEdit.css';
+
+type InlineEditProps = ComponentProps<'div'> &
+  InlineVariants &
+  Pick<ComponentProps<'input'>, 'defaultValue'> & {
+    'data-test-id'?: string;
+    onSave: Dispatch<SetStateAction<string>>;
+    hideEdit?: boolean;
+    input?: ReactElement<TextFieldProps | TextAreaProps>;
+  };
+
+const InlineEdit = ({
+  'data-test-id': testId = 'inline-edit',
+  layout = 'horizontal',
+  children,
+  defaultValue,
+  onSave,
+  hideEdit = false,
+  input = <TextField />,
+  'aria-label': ariaLabel,
+}: InlineEditProps) => {
+  const [isEditing, setEditing] = useState(false);
```

Comment:
Same question the one about `defaultValue` and un/controlled.

Are there use cases where it would be useful to pass `isEditing` from the owner component?